node/pkg/altpub/alternate_pub.go

```
@@ -0,0 +1,534 @@
+	if !ok {
+		return nil, errors.New("failed to cast default transport for alternate publisher")
+	}
+
+	t := defTrans.Clone()
+	t.MaxIdleConns = MaxIdleConns
+	t.MaxConnsPerHost = MaxConnsPerHost
+	t.MaxIdleConnsPerHost = MaxIdleConnsPerHost
+
+	return &http.Client{
+		Timeout:   HttpClientTimeout,
+		Transport: t,
+	}, nil
+}
+
+// startHttpWorkers starts all of the workers in the HTTP worker pool.
+func (ap *AlternatePublisher) startHttpWorkers(ctx context.Context, client *http.Client, errC chan error, numWorkers int) {
+	for count := range numWorkers {
+		worker := &HttpWorker{ap, client, count}
+		common.RunWithScissors(ctx, errC, "alt_pub_worker", worker.httpWorker)
+	}
+}
+
+// httpWorker is the entrypoint for an HTTP worker. Each httpWorker is responsible for posting an HTTP request and waiting for the response.
+func (w *HttpWorker) httpWorker(ctx context.Context) error {
+	for {
+		select {
+		case <-ctx.Done():
+			return nil
+		case req := <-w.ap.httpWorkerChan:
+			w.ap.httpPost(ctx, w.client, req)
+		}
+	}
+}
+
+// httpPost actually posts an HTTP request and waits for the response. It pegs metrics based on the results.
+func (ap *AlternatePublisher) httpPost(ctx context.Context, client *http.Client, req *HttpRequest) {
+	channelDelay.WithLabelValues(req.ep.label).Observe(float64(time.Since(req.start).Microseconds()))
+
+	// Create the HTTP POST request using our context so it can be interrupted.
+	// Note that we are not using a timeout context because the HTTP client already has a timeout.
+	// Note that we make a copy of the payload because `bytes.NewBuffer` takes ownership of the data, and the same data might be in multiple requests.
+	r, err := http.NewRequestWithContext(ctx, "POST", req.url, bytes.NewBuffer(bytes.Clone(req.data)))
+	if err != nil {
+		ap.logger.Error("Failed to create post request", zap.Error(err))
+		requestFailed.WithLabelValues(req.ep.label, "create_failed").Inc()
+		return
```

Why not return the errors in this function?